internal/expr/functions/definition.go

```diff
@@ -57,13 +57,20 @@ func (fd *definition) Name() string {
 }
 
 func (fd *definition) Function(args ...expr.Expr) (expr.Function, error) {
+	if fd.arity == -1 {
+		return fd.constructorFn(args...)
+	}
+
 	if len(args) != fd.arity {
 		return nil, fmt.Errorf("%s() takes %d argument(s), not %d", fd.name, fd.arity, len(args))
 	}
 	return fd.constructorFn(args...)
 }
 
 func (fd *definition) String() string {
+	if fd.arity < 0 {
+		fd.arity = 0
```

Comment:
I wouldn't change the arity here. Instead we could use a local variable
```go
arity := fd.arity
if arity < 0 {
...
```

Reply:
fixed